#!/usr/bin/env python
PACKAGE = "uavcan_core"
MAX_FREQ = 200

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("current_pid",    int_t, 0, "Current PID error stream frequency",   0, 0, MAX_FREQ)
gen.add("velocity_pid",   int_t, 0, "Velocity PID error stream frequency",  0, 0, MAX_FREQ)
gen.add("position_pid",   int_t, 0, "Position PID error stream frequency",  0, 0, MAX_FREQ)
gen.add("index",          int_t, 0, "Index value stream frequency",         0, 0, MAX_FREQ)
gen.add("motor_encoder",  int_t, 0, "Encoder raw encoder stream frequency", 0, 0, MAX_FREQ)
gen.add("motor_state",    int_t, 0, "State stream frequency",               0, 0, MAX_FREQ)

exit(gen.generate(PACKAGE, "uavcan_core", "FeedbackStream"))
